### PR TITLE
Microservice log centralization with logstash

### PR DESCRIPTION
**Pull request by PierreBesson**
Hi guys,
This is something I have been working on for a long time. Initially I wanted to do the config in logback-spring.xml but then there will be no enable/disable switch like we currently have for metrics. The choice of forwarding logs directly to a UDP socket with logback instead of through console output has also been carefully considered.

To try it out see: [jhipster/jhipster-registry/pull/15](https://github.com/jhipster/jhipster-registry/pull/15)
[EDIT] Corrected link

### CODE REVIEW

**Comment by deepu105** [comment]
tests failing are from master and not this PR

File: generators/server/templates/src/main/java/package/config/_JHipsterProperties.java [hunk 1]
```diff
@@ -514,6 +514,47 @@ public void setPrefix(String prefix) {
...
+
+        public Logstash getLogstash() { return logstash; }
+
+        public static class Logstash {
+
+            private boolean enabled = false;
+
+            private String host = "localhost";
+
+            private int port = 512;
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/config/_JHipsterProperties.java]: maybe 5000 by default, instead ?

**Comment by PierreBesson** [comment]
I corrected the port to 5000 and pushed. I hope the tests will pass this time.

**Comment by PierreBesson** [comment]
It seems the upgrade to spring boot 1.3.3 broke this feature. When I switch the spring boot version in the pom to 1.3.2 it works. This is related to the dependency on [github.com/logstash/logstash-logback-encoder](https://github.com/logstash/logstash-logback-encoder). I opened an issue on their github about this.

**Comment by jdubois** [comment]
@PierreBesson then the quick & easy solution for us: fix the dependency in the pom.xml (and Gradle!). We already do this for a number of other dependencies, no problem with that.

**Comment by PierreBesson** [comment]
@jdubois Adding the exclusions did the trick. It is also more future proof. Thanks for the advice.

**Comment by PierreBesson** [comment]
This is finally passing !

**Pull request merged**